search/query_language_whitebox_test.go

```diff
@@ -270,7 +271,76 @@ func TestParseMap(t *testing.T) {
 		assert.Equal(t, expectedQuery, actualQuery)
 	})
 
+	t.Run(OPTS, func(t *testing.T) {
+		t.Parallel()
+		// given
+		input := fmt.Sprintf(`{"%s": [ {"parent-exists": true}, {"tree-view": true}]}`, OPTS)
+		// Parsing/Unmarshalling JSON encoding/json
+		fm := map[string]interface{}{}
+		err := json.Unmarshal([]byte(input), &fm)
+		require.Nil(t, err)
+		// when
+		actualQuery := Query{}
+		parseOptions(fm, &actualQuery)
+		// then
+		qo := QueryOptions{ParentExists: true, TreeView: true}
+		expectedQuery := Query{Options: &qo}
```

Comment:
Simple, but please just create the options inside the Query.

Reply:
The `parseOptions` takes the whole query which includes options and rest of the query. However in this particular test case, the input string only has options. The top-level query object will contain the options.  The test case for `parseFilterString` demonstrate that.

This particular line (287) is required to make assertion that options inside the `Query` object. However this query object is not what used when parsing the whole string.  The same string will be parsed two time.  The first time for query without options and second time only for options.  The query objected created from the first instance is modified to set the options as an attribute which we get after parsing for the second time.

Reply:
@baijum I don't get it. Where's the difference between

```go
qo := QueryOptions{ParentExists: true, TreeView: true}
expectedQuery := Query{Options: &qo}
```

and my proposed shorter version:

```go
expectedQuery := Query{Options: &QueryOptions{ParentExists: true, TreeView: true}}
```
?

Reply:
Sorry, I misunderstood your original question.

That makes sense, I will change it.